apps/hash-graph/lib/graph/src/snapshot/entity/batch.rs

```diff
@@ -0,0 +1,193 @@
+use async_trait::async_trait;
+use error_stack::{IntoReport, Result, ResultExt};
+use tokio_postgres::GenericClient;
+
+use crate::{
+    snapshot::{
+        entity::{EntityEditionRow, EntityIdRow, EntityLinkEdgeRow, EntityTemporalMetadataRow},
+        WriteBatch,
+    },
+    store::{AsClient, InsertionError, PostgresStore},
+};
+
+pub enum EntityRowBatch {
+    Ids(Vec<EntityIdRow>),
+    Editions(Vec<EntityEditionRow>),
+    TemporalMetadata(Vec<EntityTemporalMetadataRow>),
+    Links(Vec<EntityLinkEdgeRow>),
+}
+
+#[async_trait]
+impl<C: AsClient> WriteBatch<C> for EntityRowBatch {
+    async fn begin(postgres_client: &PostgresStore<C>) -> Result<(), InsertionError> {
+        postgres_client
+            .as_client()
+            .client()
+            .simple_query(
+                r"
+                    CREATE TEMPORARY TABLE entity_ids_tmp
+                        (LIKE entity_ids INCLUDING ALL)
+                        ON COMMIT DROP;
+
+                    CREATE TEMPORARY TABLE entity_editions_tmp (
+                        entity_edition_id UUID PRIMARY KEY,
+                        properties JSONB NOT NULL,
+                        left_to_right_order INT4,
+                        right_to_left_order INT4,
```

Comment:
It's nice to have the size of the int in the type name here, but after looking at the docs, it seems that `integer` is more standard to SQL https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-INT

Reply:
Yeah my immediate thought was to use INT4 in the migrations too, but after checking the docs it seems non-standard :sweat_smile: